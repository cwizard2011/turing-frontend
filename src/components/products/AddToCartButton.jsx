import React from 'react';
import PropTypes from 'prop-types';

const AddToCartButton = ({ handleAddToCart, color, size }) => (
  <div>
    <button
      type="button"
      className="register-button mt-4"
      onClick={handleAddToCart}
      disabled={
          color === null
          || color === 'Select color'
          || size === null
          || size === 'Select size'
      }
    >
        Add to cart
    </button>
  </div>
);

AddToCartButton.propTypes = {
  handleAddToCart: PropTypes.func,
  color: PropTypes.string,
  size: PropTypes.string
};

export default AddToCartButton;
