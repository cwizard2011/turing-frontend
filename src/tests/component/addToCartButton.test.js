import React from 'react';
import { shallow } from 'enzyme';
import AddToCartButton from '../../components/products/AddToCartButton';


describe('Editor component', () => {
  it('should render View Products correctly', () => {
    const wrapper = shallow(<AddToCartButton />);
    expect(wrapper).toMatchSnapshot();
  });
});
