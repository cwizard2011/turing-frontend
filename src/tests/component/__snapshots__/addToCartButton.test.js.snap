// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Editor component should render View Products correctly 1`] = `
<div>
  <button
    className="register-button mt-4"
    disabled={false}
    type="button"
  >
    Add to cart
  </button>
</div>
`;
